shared/macros/10-warning.jinja

```diff
@@ -23,11 +23,17 @@
 :type rule: str
 :param release: Release number since the rule is officially deprecated.
 :type release: str
+:param is_rule: By default reason of deprecation is newer rule, but can be other circumstances
+:type is_rule: bool
 
 #}}
-{{% macro warning_rule_deprecated_by(rule, release='') %}}
+{{% macro warning_rule_deprecated_by(cause, release='', is_rule=True) %}}
```

Comment:
Maybe it would be easier to understand this parameter if we call it `rule_or_reason`, for example. The parameters names and respective descriptions should also be updated above.